luci-app-wireguard: add description

rwalli opened the pull request:
Add description filed to luci-app-wireguard as in luci-proto-wireguard:
https://github.com/openwrt/luci/commit/140f41b40f54feb4fb8d6ddbbdf3c659c779756d#diff-daba76e1569c21be9540dd6307bd578d

This needs reviewing because I don't know lua.

jow- commented:
Please fold all changes into one commit.

jow- commented on applications/luci-app-wireguard/luasrc/view/wireguard.htm:
```diff
@@ -179,10 +179,26 @@ <h2>WireGuard Status</h2>
+      local tmp_desc, pub_key = "", ""
+      for key, value in pairs(s) do
+        if key == "description" then
+          tmp_desc = value;
```
No semicolons please

jow- commented on applications/luci-app-wireguard/luasrc/view/wireguard.htm:
```diff
@@ -179,10 +179,26 @@ <h2>WireGuard Status</h2>
+    local desc, tmp_desc, pub_key = "", "", ""
+    cur:foreach("network", "wireguard_" .. ikey, function(s)
+      local tmp_desc, pub_key = "", ""
+      for key, value in pairs(s) do
```
Please add `local key, value`

jow- commented on applications/luci-app-wireguard/luasrc/view/wireguard.htm:
```diff
@@ -179,10 +179,26 @@ <h2>WireGuard Status</h2>
     -%>
     <div class="tr">
-      <div class="td" width="33%" style="vertical-align:top"><%:Peer%></div>
+      <div class="td" width="33%" style="vertical-align:top"><%:Peer%><%=desc%></div>
```
Change this into `<%=pcdata(desc)%>` to avoid XSS

rwalli commented:
Thank you for your suggestions!

I changed your suggestions locally and squashed all 4 commits into one.
Unfortunately something went wrong and I could not push my changes.

Sorry for inconveniences!

What should I do now?
close this PR?

hnyman commented:
> squashed all 4 commits into one.
> Unfortunately something went wrong and I could not push my changes.

Note that you need to use "git push -f" to force push this PR feature branch (patch-1). Your new squashed commit will overwrite the original ones at Github, so you need to force it.

See. e.g. https://github.com/openwrt/luci/blob/master/CONTRIBUTING.md#advice-on-pull-requests

rwalli commented:
I've read several "howto's" including the link you suggested and could not push my local repo.
I really tried many hours, Sorry!

jow- commented:
Superseded by 0fe2e72f7
